sklego/linear_model.py

```diff
@@ -18,6 +18,41 @@
 )
 
 
+class ProbWeightRegression(BaseEstimator, RegressorMixin):
+    def __init__(self, min_zero=True, fit_intercept=True):
+        self.min_zero = min_zero
+        self.fit_intercept = fit_intercept
+
+    def _handle_intercept(self, X):
+        if self.fit_intercept:
+            ones = np.ones((X.shape[0], 1))
+            X = np.hstack([ones, X])
+        return X
+
+    def fit(self, X, y):
+        X, y = check_X_y(X, y, estimator=self, dtype=FLOAT_DTYPES)
+        X = self._handle_intercept(X)
+
+        # Construct the problem.
+        betas = cp.Variable(X.shape[1])
+        objective = cp.Minimize(cp.sum_squares(X * betas - y))
+        constraints = [sum(betas) == 1]
```

Comment:
That is ... a great question. It feels as if this needs to be a hyperparameter. Strong opinions on this?

Reply:
Mhm ... is there ever a scenario ... when we want the weights to be strictly [0-1] and sum(1) while there is a constant in there? 

I am starting to feel like `fit_intercept` should be `False` in pretty much all cases. I think I added the `fit_intercept` in there in order to adhere to the scikit-learn standard but it feels like a really strange usecase.

Reply:
I can't think of a use for a non-zero intercept at the moment, but there might be some indeed. Some of the options we have:

- keep intercept but set it to False by default -> does make sense but is a strong deviation from what other scikit-learn models do
- keep intercept and set it to True by default -> is more expected due to scikit-learns default but might not make sense in this implementation
- remove intercept altogether from the model. -> probably the right thing in most cases, but we might be missing a use case

In case we do keep the intercept I do believe it should not be part of the constraints.

Reply:
I prefer to remove the feature for now and add it once there is a good, document-able, use-case for it. Otherwise the API might suggest a bad practice as a considerable option.